tofhir-engine/src/main/scala/io/tofhir/engine/mapping/FhirMappingJobManager.scala

```diff
@@ -270,10 +266,10 @@ class FhirMappingJobManager(
                                        identityServiceSettings: Option[IdentityServiceSettings] = None,
                                        timeRange: Option[(LocalDateTime, LocalDateTime)] = None,
                                        executionId: Option[String] = None
-                                      ): Future[Dataset[FhirMappingResult]] = {
+                                      ): Future[Dataset[FhirMappingResult]] = Future {
```

Comment:
Why not the following?

= {
    val (fhirMapping, mds, df) = readJoinSourceData(task, sourceSettings, timeRange, jobId = Some(jobId))
    executeTask(jobId, fhirMapping, df, mds, terminologyServiceSettings, identityServiceSettings, executionId)
  }

Reply:
executeTask returns a Future but readJoinSourceData does not and this prevent us from accessing the errors encountered in readJoinSourceData. Therefore, the logging was not working as expected. By converting it to a Future, I managed to catch errors and handle the logging properly.

Reply:
I see, thanks for the explanation.

I recommend to explain this also as a comment alongside the code.

If sth. like the following also OK, I would recommed the restructure accordingly:

    Future.apply(readJoinSourceData(task, sourceSettings, timeRange, jobId = Some(jobId))) flatMap {
      case (fhirMapping, mds, df) => executeTask(jobId, fhirMapping, df, mds, terminologyServiceSettings, identityServiceSettings, executionId)
    }